stablehlo/dialect/VhloOps.td

```diff
@@ -0,0 +1,937 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+   Copyright 2022 The StableHLO Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef STABLEHLO_COMPATIBILITY_VHLO_OPS
+#define STABLEHLO_COMPATIBILITY_VHLO_OPS
+
+include "mlir/Dialect/Shape/IR/ShapeBase.td"
+include "mlir/IR/OpBase.td"
+
+//===----------------------------------------------------------------------===//
+// Dialect and Ops
+//===----------------------------------------------------------------------===//
+
+def VHLO_Dialect : Dialect {
+  let name = "vhlo";
+  let cppNamespace = "::mlir::vhlo";
+
+  let description = [{
+    A shim opset of Versioned StableHLO ops for versions 0.x.x and 1.x.x.
+    
+    Version log:
+      0.0.0: Bootstrap from MHLO: https://github.com/openxla/stablehlo/pull/1.
+      0.1.0: Add CustomCallOp input output aliasing: https://github.com/openxla/stablehlo/pull/403
+  }];
+
+  let useDefaultAttributePrinterParser = 0;
+  let useDefaultTypePrinterParser = 0;
+}
+
+include "stablehlo/dialect/VhloEnums.td"
+include "stablehlo/dialect/VhloAttrs.td"
+
+// Most ops should not use traits. Exceptions are:
+// - ReturnOp needs a trait for Terminator.
+// - ReduceOp/ReduceWindowOp/ScatterOp need a trait since they have
+//   multiple variadic arguments.
+class VHLO_Op<string mnemonic, string minVersion = "0.0.0", string maxVersion = "current", list<Trait> traits = []> :
+    Op<VHLO_Dialect, mnemonic, 
+      [DeclareOpInterfaceMethods<VersionedInterface>] # traits> {
+  let extraClassDefinition = [{
+    mlir::vhlo::Version $cppClass::getMinVersion() {
+      return *mlir::vhlo::Version::get("}] #  minVersion # [{");
+    }
+    mlir::vhlo::Version $cppClass::getMaxVersion() {
+      if (!strcmp("}] # maxVersion # [{", "current")) return VhloDialect::getCurrentDialectVersion();
+      return *mlir::vhlo::Version::get("}] #  maxVersion # [{");
+    }
+  }];
+}
+
+def VHLO_ConstantOpV1 : VHLO_Op<"constant"> {
```

Comment:
Let's also rename mnemonics to follow the op names? If this is a lot of work, let's defer the discussion and potential execution to a ticket.

Reply:
My initial design had `_v1` mnemonics and we decided to avoid that design because was more verbose without much benefit. The notion of un-suffixed operations being v1 exists in TF Lite, for example.